tests/integration/test_bundle.py

```diff
@@ -0,0 +1,139 @@
+    await ops_test.model.applications["knative-serving"].set_config(
+        {
+            "istio.gateway.name": ISTIO_GATEWAY_NAME,
+            "istio.gateway.namespace": ops_test.model.name,
+        }
+    )
+    await ops_test.model.wait_for_idle(status="active", timeout=1000, idle_period=90, raise_on_error=False)
+
+
+def test_inference_service_serverless_deployment(
+    ops_test: OpsTest, lightkube_client: lightkube.Client
+):
+    """Create an InferenceService and validate it has a status."""
+    # Use the model namespace for deploying the ISVC
+    serverless_namespace = ops_test.model.name
+
+    # Create InferenceService from example file
+    @tenacity.retry(
+        wait=tenacity.wait_exponential(multiplier=1, min=1, max=15),
+        stop=tenacity.stop_after_delay(30),
+        reraise=True,
+    )
+    def create_inf_svc():
+        lightkube_client.create(SKLEARN_INF_SVC_OBJECT, namespace=serverless_namespace)
+
+    # Assert InferenceService state is Available
+    @tenacity.retry(
+        wait=tenacity.wait_exponential(multiplier=1, min=1, max=15),
+        stop=tenacity.stop_after_attempt(30),
+        reraise=True,
+    )
+    def assert_inf_svc_state():
+        inf_svc = lightkube_client.get(
+            ISVC, SKLEARN_INF_SVC_NAME, namespace=serverless_namespace
+        )
+        conditions = inf_svc.get("status", {}).get("conditions")
+        for condition in conditions:
+            if condition.get("status") == "False":
+                status_overall = False
+                break
+            status_overall = True
+        assert status_overall is True
+
+    create_inf_svc()
+    assert_inf_svc_state()
+
+
+def test_perfom_inference(ops_test: OpsTest, lightkube_client: lightkube.Client):
+    """Perform a POST request with data for sklearn-iris ISVC."""
+    # This input data is hardcoded based on the example in https://kserve.github.io/website/latest/get_started/first_isvc/
+    sklearn_iris_input = dict(instances=[[6.8, 2.8, 4.8, 1.4], [6.0, 3.4, 4.5, 1.6]])
+    headers = {"Content-Type": "application/json"}
+    url = get_isvc_url(
+        isvc_name=SKLEARN_INF_SVC_NAME,
+        isvc_namespace=ops_test.model.name,
+        lightkube_client=lightkube_client,
+    )
+    inference_response = requests.post(
+        f"{url}/v1/models/{SKLEARN_INF_SVC_NAME}:predict",
+        headers=headers,
+        data=json.dumps(sklearn_iris_input),
+    ).text
+    assert inference_response == '{"predictions":[1,1]}'
+
+
+@tenacity.retry(
+    wait=tenacity.wait_exponential(multiplier=1, min=1, max=15),
+    stop=tenacity.stop_after_attempt(30),
+    reraise=True,
+)
+def get_isvc_url(
+    isvc_name: str, isvc_namespace: str, lightkube_client: lightkube.Client
+) -> str:
+    """Return the ISVC url gotten from an existing ISVC in the Kubernetes deployment."""
+    isvc_object = lightkube_client.get(ISVC, isvc_name, namespace=isvc_namespace)
+    return isvc_object.get("status")["components"]["predictor"]["url"]
```

Comment:
Should we define this inside the `test_perfom_inference` similar to what we do for `test_inference_service_serverless_deployment`?

Reply:
I think that this is more reusable outside of the `test_perfom_inference()` test case if we decide to extend the tests. Do you think we should put it inside of the test case just to keep consistency?

Reply:
What you describe makes sense actually.